src/MoBi.Presentation/Mappers/ModuleConfigurationToModuleConfigurationDTOMapper.cs

```diff
@@ -8,12 +10,32 @@ public interface IModuleConfigurationToModuleConfigurationDTOMapper : IMapper<Mo
    {
 
    }
-   
+
    public class ModuleConfigurationToModuleConfigurationDTOMapper : IModuleConfigurationToModuleConfigurationDTOMapper
    {
+      private readonly IMoBiContext _context;
+
+      public ModuleConfigurationToModuleConfigurationDTOMapper(IMoBiContext context)
+      {
+         _context = context;
+      }
+
       public ModuleConfigurationDTO MapFrom(ModuleConfiguration moduleConfiguration)
       {
-         return new ModuleConfigurationDTO(moduleConfiguration);
+         return new ModuleConfigurationDTO(createProjectModuleConfiguration(moduleConfiguration));
+      }
+
+      private ModuleConfiguration createProjectModuleConfiguration(ModuleConfiguration moduleConfiguration)
+      {
+         // If the module configuration already contains project building blocks then use it
+         var module = _context.CurrentProject.Modules.First(x => Equals(x.Name, moduleConfiguration.Module.Name));
+         if (module.Equals(moduleConfiguration.Module))
+            return moduleConfiguration;
+
+         // Otherwise create a new module configuration with the same name and the building blocks from the project
+         return new ModuleConfiguration(module,
+            moduleConfiguration.SelectedMoleculeStartValues == null ? null : module.MoleculeStartValuesCollection.FindByName(moduleConfiguration.SelectedMoleculeStartValues?.Name),
```

Comment:
? not needed at the end

Reply:
I would rewrite this to be more concise
module.MoleculeStartValuesCollection.FindByName(moduleConfiguration.SelectedMoleculeStartValues?.Name)??null